### PR TITLE
fix ocp version

### PR DESCRIPTION
**Pull request by ldpliu**

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/distributionInfo_sync.go [hunk 1]
```diff
@@ -77,7 +79,10 @@ func (s *distributionInfoSyncer) syncOCPDistributionInfo(ctx context.Context, cl
 		history.Verified = historyItem.Verified
 		ocpDistributionInfo.VersionHistory = append(ocpDistributionInfo.VersionHistory, history)
 		if historyItem.State == "Completed" {
```
Comment (zhiweiyin318) [inline pkg/klusterlet/clusterinfo/distributionInfo_sync.go:81]: could we get ocp version from clusterClaim here?

File: pkg/klusterlet/clusterclaim/clusterclaimer.go [hunk 2]
```diff
@@ -338,10 +338,14 @@ func (c *ClusterClaimer) getOCPVersion() (version, clusterID string, err error)
...
 	historyItems := clusterVersion.Status.History
+	var latestCompleteTime *metav1.Time
+
 	for _, historyItem := range historyItems {
 		if historyItem.State == "Completed" {
-			version = historyItem.Version
-			break
+			if latestCompleteTime.IsZero() || latestCompleteTime.Before(historyItem.CompletionTime) {
+				latestCompleteTime = historyItem.CompletionTime
+				version = historyItem.Version
```
Comment (zhujian7) [inline pkg/klusterlet/clusterclaim/clusterclaimer.go:347]: Could we rename the v`ersion` to `latestCompleteVersion` to match the `latestCompleteTime`

File: pkg/klusterlet/clusterinfo/distributionInfo_sync.go [hunk 3]
```diff
@@ -76,10 +79,15 @@ func (s *distributionInfoSyncer) syncOCPDistributionInfo(ctx context.Context, cl
...
 		history.Verified = historyItem.Verified
 		ocpDistributionInfo.VersionHistory = append(ocpDistributionInfo.VersionHistory, history)
-		if historyItem.State == "Completed" {
-			ocpDistributionInfo.Version = historyItem.Version
-		}
 	}
+
+	version, err := s.claimLister.Get(clusterclaim.ClaimOpenshiftVersion)
+	if err != nil {
+		klog.Errorf("Failed to get ocp version, error: %v", err)
```
Comment (zhujian7) [inline pkg/klusterlet/clusterinfo/distributionInfo_sync.go:86]: should we return the err?

File: pkg/klusterlet/clusterclaim/clusterclaimer.go (hunk 2 above)
Comment (ldpliu) [reply pkg/klusterlet/clusterclaim/clusterclaimer.go]: done

File: pkg/klusterlet/clusterinfo/distributionInfo_sync.go (hunk 3 above)
Comment (ldpliu) [reply pkg/klusterlet/clusterinfo/distributionInfo_sync.go:86]: > should we return the err?

I do not think it's necessary to return err. As other fields should also be updated.

Comment (zhujian7) [reply pkg/klusterlet/clusterinfo/distributionInfo_sync.go:86]: think about when will return an error here? if we can not get claim(except for the not found error), can other places get the claim successfully?
IMO we should return the error if it is not a `NotFound` error.

**Comment by zhujian7** [comment]
LGTM

**Pull request merged**